### ISSUE TITLE
Generic Git Repository ( GitLab CE ) module update issue

### ISSUE THREAD

**Issue by jare19** [hashicorp/terraform#4004]
Now, this _could_ be a bug or it _could_ be a miss-use by me but I am leaning toward bug...
I use GitLab CE for my internal git repo and am having an issue running `terraform get -update` to update a downloaded module.
After I commit the change to git I run `terraform get -update` and it does not error and acts like it has updated the changes.  I then navigate to the `.terraform/modules/IDENTIFIER` path and the new file I have committed is not there.  I have also tried editing a current/known .tf file, commiting to git, and the `terraform get -update` still reports that it is working but indeed the change is not reflected in the `.terraform/modules/IDENTIFIER` path.
The only way I can resolve this issue is by deleting the `.terraform/` dir and running `terraform get`
I don't think this is by design so any help/guidance would be appreciated.

the git source path I am using in my module.tf config file looks like this
`source = "git::http://server.gitlab.domain.net/path/terraform-mesos.git?ref=Demo"`

**Comment by jare19** [comment]
updating as I have found more information.  This problem only happens when a branch other than master is populated.

**Comment by jen20** [comment]
Thanks @jare19, I'm going to investigate this now.

**Comment by oillio** [comment]
I am seeing the same behavior with my git repository as well.  My entry look like this (slightly altered):

source = "git::ssh://git@github.com/Comapny/project-repo//deploy/terraform"

**Comment by cprenzberg** [comment]
I am seeing this issue too. Reported this yesterday at #8256

**Comment by mitchellh** [comment]
I tried reproducing both @oillio and @jare19's example cases. I tried all of the following:

  * GitHub and GitLab private repos
  * Master and non-master branches (?ref specified and not specified)
  * Root and subdirectory with "//"
  * SSH protocol 

I wasn't able to reproduce the issue anymore. The initial `terraform get` would work as well as the follow up `terraform get -update` (I would make changes in the repo then verify the changes in Terraform). 

I'm going to mark this as closed unless it can be reproduced anymore!
